Aztec/Classes/Public/Formatters/AttributeFormatter.swift

```diff
@@ -0,0 +1,148 @@
+    /// do that. Otherwise, it will toggle the attributes in the text view's
+    /// `typingAttributes` property.
+    ///
+    /// The application range might be different than the passed range, as
+    /// explained in `applicationRange(forRange:inString:)`
+    ///
+    func toggleAttribute(inTextView textView: UITextView, atRange range: NSRange)
+}
+
+extension AttributeFormatter {
+    /// Checks if the attribute is present in a text view at the specified index.
+    ///
+    func attribute(inTextView textView: UITextView, at index: Int) -> Bool {
+        let string = textView.textStorage
+        let range = applicationRange(forRange: NSRange(location: index, length: 0), inString: string)
+        let attributes = range.length > 0
+            ? string.attributesAtIndex(index, effectiveRange: nil)
+            : textView.typingAttributes
+        return present(inAttributes: attributes)
+    }
+}
+
+// MARK: - Default implementations
+
+extension AttributeFormatter {
+    func applicationRange(forRange range: NSRange, inString string: NSAttributedString) -> NSRange {
+        return range
+    }
+}
+
+// MARK: - Private methods
+
+private extension AttributeFormatter {
+    func toggleTypingAttribute(inTextView textView: UITextView) {
+        if present(inAttributes: textView.typingAttributes) {
+            removeTypingAttribute(fromTextView: textView)
+        } else {
+            addTypingAttribute(toTextView: textView)
+        }
+    }
+
+    func addTypingAttribute(toTextView textView: UITextView) {
+        for (key, value) in attributes {
+            textView.typingAttributes[key] = value
+        }
+    }
+
+    func removeTypingAttribute(fromTextView textView: UITextView) {
+        for (key, _) in attributes  {
+            textView.typingAttributes.removeValueForKey(key)
+        }
+    }
+
+    func toggleAttribute(inString string: NSMutableAttributedString, atRange range: NSRange) {
+        if attribute(inString: string, at: range.location) {
+            removeAttributes(fromString: string, atRange: range)
+        } else {
+            applyAttributes(toString: string, atRange: range)
+        }
+    }
+
+    func applyAttributes(toString string: NSMutableAttributedString, atRange range: NSRange) {
+        string.addAttributes(attributes, range: range)
+    }
+
+    func removeAttributes(fromString string: NSMutableAttributedString, atRange range: NSRange) {
+        for (attribute, _) in attributes {
+            string.removeAttribute(attribute, range: range)
+        }
+    }
+
+    func attribute(inString string: NSAttributedString, at index: Int) -> Bool {
+        let attributes = string.attributesAtIndex(index, effectiveRange: nil)
+        return present(inAttributes: attributes)
+    }
+
+    func insertEmptyAttribute(inString string: NSMutableAttributedString, at index: Int) {
+        let attributedSpace = NSAttributedString(string: "\u{200B}", attributes: attributes)
```

Comment:
I'd suggest making this a constant.  I'm not sure what the definitive location for that constand would be, but a good starting place could be as a member of this class (`AttributeFormatter`).